src/components/dateTimePicker/index.tsx

```diff
@@ -108,79 +108,66 @@ export interface DateTimePickerProps {
   testID?: string;
 }
 
-interface DateTimePickerState {
-  prevValue?: Date;
-  value?: Date;
-}
-
 type DateTimePickerPropsInternal = DateTimePickerProps & BaseComponentInjectedProps;
 
-class DateTimePicker extends Component<DateTimePickerPropsInternal, DateTimePickerState> {
-  static displayName = 'DateTimePicker';
-
-  static defaultProps = {
-    ...TextField.defaultProps,
-    mode: MODES.DATE
-  };
-
-  chosenDate?: Date;
-  expandable = React.createRef<ExpandableOverlayMethods>();
+function DateTimePicker(props: DateTimePickerPropsInternal) {
+  const {
+    renderInput,
+    editable,
+    mode = MODES.DATE,
+    dateFormat,
+    timeFormat,
+    dateFormatter,
+    timeFormatter,
+    minimumDate,
+    maximumDate,
+    locale,
+    is24Hour,
+    minuteInterval,
+    timeZoneOffsetInMinutes,
+    themeVariant,
+    onChange,
+    dialogProps,
+    headerStyle,
+    // @ts-expect-error
+    useCustomTheme,
+    testID,
+    ...others
+  } = props;
 
-  constructor(props: DateTimePickerPropsInternal) {
-    super(props);
-    this.chosenDate = props.value;
+  const [value, setValue] = useState(props.value);
```

Comment:
I think it will be more readable and less prone to bugs if you extract `value` and rename it `propsValue` instead of using it as `props.value`.